app/gui/view/component-browser/searcher-list-panel/src/lib.rs

```diff
@@ -0,0 +1,662 @@
+        ));
+        self.scroll_area.set_position_xy(Vector2::new(
+            -style.content.size.x / 2.0,
+            style.content.size.y / 2.0 - style.menu.height / 2.0,
+        ));
+        self.scroll_area.set_corner_radius_bottom_right(style.content.corner_radius);
+    }
+
+    fn recompute_layout(&self, style: &Style) {
+        self.update_style(style);
+        self.update_layers();
+
+        let favourites_section_height = self.favourites_section.height(style);
+        let local_scope_height = self.local_scope_section.height(style);
+        let sub_modules_height = self.sub_modules_section.height(style);
+
+        self.favourites_section.set_base_position_y(0.0, style);
+        self.local_scope_section.set_base_position_y(-favourites_section_height, style);
+        self.sub_modules_section
+            .set_base_position_y(-favourites_section_height - local_scope_height, style);
+
+        let full_height = favourites_section_height + local_scope_height + sub_modules_height;
+        self.scroll_area.set_content_height(full_height);
+        self.scroll_area.jump_to_y(full_height);
+    }
+}
+
+impl display::Object for Model {
+    fn display_object(&self) -> &display::object::Instance {
+        &self.display_object
+    }
+}
+
+impl component::Model for Model {
+    fn label() -> &'static str {
+        "ComponentBrowserPanel"
+    }
+
+    fn new(app: &Application, _logger: &DefaultWarningLogger) -> Self {
+        Self::new(app)
+    }
+}
+
+
+
+// =======================
+// === Labeled Section ===
+// =======================
+
+/// Struct that contains the components contained in a section of the Component Browser Panel. Also
+/// provides some utility functions for shape and layout handling.
+#[derive(Clone, Debug)]
+struct LabeledSection<T> {
+    pub label:   text::Area,
+    pub divider: hline::View,
+    pub content: T,
+}
+
+impl<T: CloneRef> CloneRef for LabeledSection<T> {
+    fn clone_ref(&self) -> Self {
+        LabeledSection {
+            label:   self.label.clone_ref(),
+            divider: self.divider.clone_ref(),
+            content: self.content.clone_ref(),
+        }
+    }
+}
+
+type WideSection = LabeledSection<component_group::wide::View>;
+type ColumnSection = LabeledSection<column_grid::ColumnGrid>;
+
+impl<T: CloneRef> LabeledSection<T> {
+    pub fn new(content: T, app: &Application) -> Self {
+        let logger = Logger::new("LabeledSection");
+        let label = text::Area::new(app);
+        let divider = hline::View::new(logger);
+        Self { label, divider, content }
+    }
+
+    fn set_style(&self, style: &Style) {
+        self.divider.size.set(Vector2(INFINITE, style.section.divider_height));
+        self.label.set_default_color(style.section.heading.color);
+        self.label.set_default_text_size(style.section.heading.size);
+        self.label.set_font(style.section.heading.font.clone());
+        self.label.set_position_y(-0.75 * style.section.heading.size.raw);
```

Comment:
magic number

Reply:
Will go away in follow-up design fix PR.